### PR TITLE
Support functionality related to editing studies through Python CLI

### PR DESCRIPTION
**Pull request by vkozareva**
This functionality was added and used in support of the U19 Broad/Cold Spring Harbor collaboration.
New modules provide support for:
* Printing existing study attributes 
* Editing study description
* Printing existing study external resources
* Deleting existing study external resources
* Creating new study external resources

These include changes in the CLI (`manage_study.py`) and the underlying API library (`scp_api.py`). Example usage has been updated to reflect new functionality. 

Other functionality (can remove if too off-target):
* `SortSparseMatrix.py` now supports compressed `.mtx.gz` files as produced by versions of Cellranger > v3. 

Some notes: 
* Module for editing study description (`edit_study_description`) currently allows raw HTML formatting through argument `accept_html` as this was needed for my use case. If this considered too risky, I can remove that functionality. 
* I have not implemented a few validation checks for the `create_study_external_resource` module, but have made a note to include it in future work.

### CODE REVIEW

File: scripts/manage_study.py [hunk 1]
```diff
@@ -223,6 +243,135 @@ def validate_metadata_file(metadata_path):
...
+    c_TOOL_STUDY_EDIT_DESC,
+    help="Edit a study description. \""
+    + args.prog
+    + " "
+    + c_TOOL_STUDY_EDIT_DESC
+    + " -h\" for more details",
+)
+parser_edit_description.add_argument(
+    '--study-name',
+    dest='study_name',
```
[the remark this answers is not in the document]
Comment (vkozareva) [reply scripts/manage_study.py]: Thanks for pointing this out! I think I was going off some of the other arguments and their explicit `dest` inclusion (like line 235). I've removed all of these and am happy to clean this up in the other parsers as well if you'd like.

**Comment by vkozareva** [comment]
Thanks for the speedy review Eric! Let me know if there's anything I missed or any other changes I should make.

File: scripts/manage_study.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (eweitz) [reply scripts/manage_study.py]: Good point -- we refined our style with `dest` and delimiters after the original code here was written.  Don't feel obliged to clean up those other places, but we also would be happy to accept such changes too.

**Comment by vkozareva** [comment]
Sorry to push up these commits after your reviews! I just wanted to remove the unnecessary `dest` inclusions in the other parsers and I also noticed that I didn't account for `dry_run` in my functions for printing study attributes and external resources!

**Pull request merged**